cg/meta/invoice.py

```diff
@@ -13,103 +23,131 @@ def __init__(self, db: Store, lims_api: LimsAPI, invoice_obj: models.Invoice):
         self.record_type = ""
         self.raw_records = []
         self._set_record_type()
+        self.genologics_lims = genologics_lims
+        self.invoice_info: Optional[InvoiceInfo] = None
 
     def _set_record_type(self):
         """Define the record_type based on the invoice object.
-        It can only be either pool or sample"""
-
+        It can only be either pool or sample."""
         if self.invoice_obj.pools:
             self.raw_records = self.invoice_obj.pools
-            self.record_type = "Pool"
+            self.record_type = RecordType.Pool
         elif self.invoice_obj.samples:
-            self.record_type = "Sample"
+            self.record_type = RecordType.Sample
             self.raw_records = self.invoice_obj.samples
 
-    def prepare_contact_info(self, costcenter):
-        """Function to prepare contact info for a customer"""
-
-        msg = (
-            f"Could not open/generate invoice. Contact information missing in database for "
-            f"customer {self.customer_obj.internal_id}. See log files."
+    def get_customer_by_cost_center(self, cost_center: str) -> Union[Customer, str]:
+        """Return the costumer based on cost center."""
+        return (
+            self.db.customer(CustomerNames.cust999)
+            if cost_center.lower() == CostCenters.kth
+            else self.customer_obj
         )
 
-        customer = self.db.customer("cust999") if costcenter.lower() == "kth" else self.customer_obj
-        user = customer.invoice_contact
-
-        if not user:
+    def get_customer_invoice_contact(self, customer: Customer, msg: str) -> Any:
+        """Return the customer invoice contact."""
+        if not customer.invoice_contact:
             self.log.append(msg)
             return None
+        return customer.invoice_contact
 
-        contact = {
-            "name": user.name,
-            "email": user.email,
-            "customer_name": customer.name,
-            "reference": customer.invoice_reference,
-            "address": customer.invoice_address,
-        }
+    def get_contact(
+        self, customer: Customer, customer_invoice_contact: Optional[User], msg: str
+    ) -> Optional[Contact]:
+        """Return the contact information."""
+        try:
+            contact = Contact(
+                **{
```

Comment:
I think you can instantiate the class using args like this example:

```
MipAnalysis(
            case=mip_config.case_id,
            genome_build=sample_info.genome_build,
            sample_id_metrics=qc_metrics.sample_id_metrics,
            mip_version=sample_info.mip_version,
            rank_model_version=sample_info.rank_model_version,
            sample_ids=mip_config.sample_ids,
            sv_rank_model_version=sample_info.sv_rank_model_version,
        )
```
If you want to use  supply an entire dict when instantiating you usually just:

```
mip_config: MipBaseConfig = MipBaseConfig(**config_raw)
```